util/src/main/java/io/kubernetes/client/informer/cache/ReflectorRunnable.java

```diff
@@ -224,7 +224,11 @@ private void watchHandler(Watchable<ApiType> watch) {
       }
       if (eventType.get() == EventType.ERROR) {
         if (item.status != null && item.status.getCode() == HttpURLConnection.HTTP_GONE) {
-          log.info("Watch connection expired: {}", item.status.getMessage());
+          log.info(
+              "ResourceVersion {} and Watch connection expired: {} , will retry w/o resourceVersion next time",
+              getRelistResourceVersion(),
+              item.status.getMessage());
+          isLastSyncResourceVersionUnavailable = true;
```

Comment:
@aaruna upon 410 status code, the watch handler is supposed to return and we deal w/ the RV expiration at the list call. this is working as designed. https://github.com/kubernetes/kubernetes/blob/a96000311f5beca111debc8727018e42cbb5dc79/staging/src/k8s.io/client-go/tools/cache/reflector.go#L129-L131

Reply:
Are you referring to the next list call? Because the current watch handler won't throw an ApiException and only that is handled in the catch block above. (The unit test added in this PR shows an example of this scenario)

CC @jiangzhou in case I have missed anything in the scenario above.

Reply:
> Are you referring to the next list call?

yeah. IIRC watch handler doesnt directly throw an ApiException but it will deal w/ error status nested in the decoded watch event. 
https://github.com/kubernetes-client/java/blob/c580990fa9de11a549cc1ae4219cb4fae994b9b5/util/src/main/java/io/kubernetes/client/informer/cache/ReflectorRunnable.java#L227-L237

Reply:
But that's not resetting `isLastSyncResourceVersionUnavailable=true`. The unit test below has an example